modules/@apostrophecms/migration/lib/addMissingSchemaFields.js

```diff
@@ -35,12 +35,48 @@ module.exports = (self) => {
       }, async doc => {
         const changes = {};
         await self.addMissingSchemaFieldsFor(doc, schema, '', changes);
+        if (doc.type === '@apostrophecms/user') {
+          // In an abundance of caution we should leave permissions-related
+          // properties alone. The role will only be missing if transitioning
+          // from advanced permissions back to regular and there is a right
+          // way to do that (via a task) so we should not interfere. There
+          // should be no default for the password, but don't chance it;
+          // also leave advanced permission group data alone just for good measure
+          delete changes.password;
+          delete changes.role;
+          delete changes.groupsIds;
+          delete changes.groupsFields;
+        }
         if (Object.keys(changes).length > 0) {
           updates++;
+          const seen = new Set();
+          const keep = {};
+          for (const key of Object.keys(changes)) {
+            // MongoDB won't tolerate $set for both
```

Comment:
It seems like the existing unit tests should have caught this, so I don't know of a good way to test it better, but in testbed this did come up and required the fix I have implemented here.